app/components/sidebars/main/index.js

```diff
@@ -15,11 +16,19 @@ import {isLandscape, isTablet, getDimensions} from 'app/selectors/device';
 
 import MainSidebar from './main_sidebar.js';
 
+export const getChannelIdsInPostsInChannel = createIdsSelector(
+    (state) => state.entities.posts.postsInChannel,
+    (postsInChannel) => {
+        return Object.keys(postsInChannel);
+    }
+);
+
 function mapStateToProps(state) {
     const {currentUserId} = state.entities.users;
 
     return {
         ...getDimensions(state),
+        loadedChannelIds: getChannelIdsInPostsInChannel(state),
```

Comment:
I added this props to easily identify whether a channel has been previously loaded or not, knowing that previously loaded channel will render faster than the one which will still going to request from the server.  Just a good metrics to compare.